linux/troubleshoot

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+if [[ ${-} != *i* ]]; then
+  source_once &> /dev/null && return 0
+fi
+
+set -eu
+
+if [ -z "${VSI_COMMON_DIR+set}" ]; then
+  VSI_COMMON_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.."; pwd)"
+fi
+
+source "${VSI_COMMON_DIR}/tests/testlib.bsh"
+source "${VSI_COMMON_DIR}/linux/colors.bsh"
+
+function run_troubleshoot_file()
+{
+  if [ -r "${1}" ]; then
+  (
+    source "${1}"
+    for test_name in $(compgen -A function troubleshoot_); do
+      begin_test "${test_name}"
+      xtrace=/dev/null
+      (
+          setup_test
+          "${test_name}"
+      )
+      end_test
+      if [ "${test_status}" != "0" ]; then
+        error_name=error_${test_name#troubleshoot_}
+        "${error_name}"
+      fi
+    done
+    TESTLIB_TEST_FILENAME=$(basename "${1}") atexit
+  ) || :
+  else
+    echo "Skipping ${1}, cannot read"
```

Comment:
Is there any more information to give here? like cannot fine troubleshoot file or something

Reply:
There are multiple reasons why it cannot read. It could not exist, be a bad symlink, exit but not have read permissions. Etc. This is why I specifically test `-r` cause this tests all these corner cases.